### PR TITLE
Dependency Injection Rewrite

### LINKED ISSUE #1988: Ignore input outside area does not function
## Description

Ignore input outside area clamps input or sends it to the bottom right of the monitor area instead of ignoring it.

This does not appear to have any workaround like #1957. They are broken no matter what is done.

Reproduction steps
1. Right click the area editor
2. Enable Ignore input outside area
3. Move your pen outside your tablet area and see that it still moves the cursor
4. Optionally you can set your area to something ridiculously small like 1mm x 1mm or drag the area beyond the bounds of the tablet for easier testing.

## System Information:

| Name             | Value |
| ---------------- | ----- |
| Operating System         | Win10
| OpenTabletDriver Version | 0.6.0.2

### PR DESCRIPTION
**Pull request by InfinityGhost**
## Notes

- Completely rewrote the OpenTabletDriver Daemon and UI projects from scratch using Dependency Injection.
- Merged the OpenTabletDriver.Plugin library into the core OpenTabletDriver library.

## Issues

- Fixes #1988

### CODE REVIEW

File: OpenTabletDriver.Daemon/DriverDaemon.cs [hunk 1]
```diff
@@ -4,50 +4,83 @@
...
-            Driver.CompositeDeviceHub.DevicesChanged += async (sender, args) =>
+            _driver.InputDevicesChanged +=
+                (sender, e) => TabletsChanged?.Invoke(sender, e.Select(c => c.Configuration));
+            _deviceHub.DevicesChanged += async (_, args) =>
             {
                 if (args.Additions.Any())
                 {
                     await DetectTablets();
-                    await SetSettings(Settings);
+                    await ApplySettings(_settingsManager.Settings);
```
[the remark this answers is not in the document]
Comment (InfinityGhost) [reply OpenTabletDriver.Daemon/DriverDaemon.cs:83]: Is this actually a source of any issues that we're aware of?

File: OpenTabletDriver.UX/App.cs [hunk 2]
```diff
@@ -1,138 +1,331 @@
...
-            var root = new RootCommand("OpenTabletDriver UX")
+            var command = new RootCommand("OpenTabletDriver UX")
             {
-                new Option<bool>(new string[] { "-m", "--minimized" }, "Start the application minimized")
+                Name = "OpenTabletDriver",
+                TreatUnmatchedTokensAsErrors = true,
+                Handler = CommandHandler.Create(Invoke)
+            };
+
+            var code = command.Invoke(Arguments);
```
[the remark this answers is not in the document]
Comment (InfinityGhost) [reply OpenTabletDriver.UX/App.cs:36]: This is all on the same thread? I don't see how this is a problem.

[the remark this answers is not in the document]
Comment (InfinityGhost) [reply OpenTabletDriver.UX/App.cs:36]: Actually, using InvokeAsync here causes the application to instantly exit, that will not work.

[the remark this answers is not in the document]
Comment (AkiSakurai) [reply OpenTabletDriver.UX/App.cs:36]: > This is all on the same thread? I don't see how this is a problem.
nope
https://github.com/dotnet/command-line-api/blob/5618b2d243ccdeb5c7e50a298b33b13036b4351b/src/System.CommandLine/Invocation/InvocationPipeline.cs#L51

[the remark this answers is not in the document]
Comment (InfinityGhost) [reply OpenTabletDriver.UX/App.cs:54]: Looks like we'll have to return to the old method to make this work, unfortunately.

**Comment by gonX** [comment]
Linux lg2m and tested working well enough that a prerelease is viable.

Btw, you can add "Fixes #1988" now

**Review by AkiSakurai — COMMENTED** [review]
Some minor issues collected, it is cumbersome to recreate separated issues, I just leave the comments here.

File: OpenTabletDriver.UX/Windows/ConfigurationEditor.cs [hunk 3]
```diff
@@ -0,0 +1,728 @@
...
+
+            Width = 1000;
+            Height = 700;
+
+            var placeholder = new Placeholder("No configuration selected.");
+
+            var splitter = new Splitter
+            {
+                Panel1MinimumSize = 250,
+                Panel1 = _configsList = new ListBox<TabletConfiguration>(),
```
[the remark this answers is not in the document]
Comment (InfinityGhost) [reply OpenTabletDriver.UX/Windows/ConfigurationEditor.cs:41]: Weird that the macOS implementation doesn't have a scrollable built into that

[the remark this answers is not in the document]
Comment (AkiSakurai) [reply OpenTabletDriver.UX/Windows/ConfigurationEditor.cs:41]: Also happens on linux
Screenshot from 2022-08-22 20-07-24

**Pull request merged**